api/report_rule_notification_types.go

```diff
@@ -0,0 +1,276 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/fatih/structs"
+	"github.com/pkg/errors"
+)
+
+type ReportRuleNotification interface {
+	allNotifications() ReportRuleNotificationTypes
+	ToMap() map[string]bool
+}
+
+// Enable all Gcp report rules
+var AllGcpReportRuleNotifications = new(GcpReportRuleNotifications).allNotifications()
+
+// Enable all Aws report rules
+var AllAwsReportRuleNotifications = new(AwsReportRuleNotifications).allNotifications()
+
+// Enable all Azure report rules
+var AllAzureReportRuleNotifications = new(AzureReportRuleNotifications).allNotifications()
+
+// Enable all Daily report rules
+var AllDailyReportRuleNotifications = new(DailyEventsReportRuleNotifications).allNotifications()
+
+// Enable all Weekly report rules
+var AllWeeklyReportRuleNotifications = new(WeeklyEventsReportRuleNotifications).allNotifications()
+
+// Enable all report rules
+var AllReportRuleNotifications = new(ReportRuleNotificationTypes).allNotifications()
+
+func NewReportRuleNotification(notificationsMap map[string]bool, notificationType ReportRuleNotification) error {
```

Comment:
this function doesn't create something "new" are we transforming it, if so, change the function name